### PR TITLE
Add endIndex property to token. Resolves #38

### LINKED ISSUE #38: Add endIndex property to Token class
This needs to be calculated in raw and assembly parsers.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/parseAsm.js [hunk 1]
```diff
@@ -100,12 +100,14 @@ function parseDataLiteral (chars, idx, terminator, start = idx) {
     throw new Error(`Unterminated literal starting at position ${start}`)
   }
 
-  // remove trailing `]` or whitespace
-  idx += 1
+  // remove trailing `]` (or other non-whitespace terminator)
+  if (typeof first !== 'undefined' && !WS.has(first)) {
```
Comment (JBaczuk) [inline src/parseAsm.js]: I just noticed, should `WS` include the following?

```
' '      space 
'\t'     horizontal tab 
'\n'     newline
'\v'     vertical tab 
'\f'     feed 
'\r'     carriage return  
```

**Comment by JBaczuk** [comment]
Also should be rebased i think

**Review by JBaczuk — CHANGES_REQUESTED** [review]
just one thing about the whitespace.

File: src/parseAsm.js (hunk 1 above)
Comment (wardlem) [reply src/parseAsm.js]: Yes, I think you are right.

**Comment by wardlem** [comment]
@JBaczuk I added the whitespace characters to the parser's WS set.  I also rebased.  Please review the changes when you get the chance.

**Review by JBaczuk — APPROVED** [review]
looks good

**Pull request merged**